avr-hal-generic/src/i2c.rs

```diff
@@ -499,5 +499,82 @@ macro_rules! impl_twi_i2c {
                 Ok(())
             }
         }
+
+        $(#[$i2c_attr])*
+        pub struct Slave<CLOCK: $crate::clock::Clock, M> {
+            p: $I2C,
+            _clock: ::core::marker::PhantomData<CLOCK>,
+            sda: $sdamod::$SDA<M>,
+            scl: $sclmod::$SCL<M>,
+            address: u8,
+        }
+
+        impl<CLOCK> Slave<CLOCK, $crate::i2c::I2cPullUp>
+        where
+            CLOCK: $crate::clock::Clock,
+        {
+            /// Initialize the I2C bus
+            ///
+            /// `new()` will enable the internal pull-ups to comply with the I2C
+            /// specification.  If you have external pull-ups connected, please
+            /// use `new_with_external_pullup()` instead.
+            ///
+            /// TODO: verify whether we need the speed param (bitrate)
+            pub fn new(
+                p: $I2C,
+                sda: $sdamod::$SDA<$crate::port::mode::Input<$crate::port::mode::PullUp>>,
+                scl: $sclmod::$SCL<$crate::port::mode::Input<$crate::port::mode::PullUp>>,
+                speed: u32,
```

Comment:
I think the speed parameter can be dropped here as the bus master is responsible for the clock.